### PR TITLE
Improved key handling in swaylock

### PR DESCRIPTION
**Pull request by snaggen**
I noticed a few things with swaylock that bothered me:
1. Escape key was appended to buffer
2. No CapsLock indicator ( https://github.com/swaywm/sway/issues/1613 )
3. Key press of modifier keys had no indication, not a biggie but with the non-standard input feedback I felt it would be nice to still show that a key was pressed. 

This pull request adds CapsLock indication, made escape clear the buffer and add some more visual feedback states.

### CODE REVIEW

File: swaylock/password.c [hunk 1]
```diff
@@ -105,11 +105,32 @@ void swaylock_handle_key(struct swaylock_state *state,
...
 		case XKB_KEY_BackSpace:
 			if (backspace(&state->password)) {
 				state->auth_state = AUTH_STATE_BACKSPACE;
-				render_frames(state);
+			} else {
+				state->auth_state = AUTH_STATE_CLEAR;
 			}
+			render_frames(state);
+			break;
+		case XKB_KEY_Escape:  /* Keys to ignore */
```
Comment (emersion) [inline swaylock/password.c]: This comment seems outdated

File: swaylock/seat.c [hunk 2]
```diff
@@ -88,6 +88,7 @@ static void keyboard_modifiers(void *data, struct wl_keyboard *wl_keyboard,
 	xkb_mod_mask_t mask = xkb_state_serialize_mods(state->xkb.state,
 			XKB_STATE_MODS_DEPRESSED | XKB_STATE_MODS_LATCHED);
 	state->xkb.modifiers = 0;
+	state->xkb.caps_lock = mods_locked & 0x02;
```
Comment (emersion) [inline swaylock/seat.c]: Isn't there a constant for this one?

Comment (snaggen) [reply swaylock/seat.c]: There is actually WLR_MODIFIER_CAPS in wlr/types/wlr_keyboard.h. However, including that caused some errors due to deprecation... And besides that one, I didn't find an obvious one...

**Comment by emersion** [comment]
@snaggen Can you rebase this branch?

**Comment by snaggen** [comment]
Ok, will do in a few hours.

Den tis 24 apr. 2018 13:55emersion <notifications@github.com> skrev:

**Comment by snaggen** [comment]
Rebased!

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**